src/Workspaces/Core/Portable/Rename/ConflictResolution.cs

```diff
@@ -19,23 +19,21 @@ namespace Microsoft.CodeAnalysis.Rename
         /// If this is true, the <see cref="ErrorMessage"/> would be null. All the other fields or properties would be valid.
         /// </summary>
         [MemberNotNullWhen(false, nameof(ErrorMessage))]
-        [MemberNotNullWhen(true, nameof(_newSolutionWithoutRenamedDocument))]
-        [MemberNotNullWhen(true, nameof(_renamedDocument))]
+        [MemberNotNullWhen(true, nameof(RenamedDocument))]
         [MemberNotNullWhen(true, nameof(OldSolution))]
-        [MemberNotNullWhen(true, nameof(NewSolution))]
+        [MemberNotNullWhen(true, nameof(NewSolutionWithoutRenamedDocument))]
         public bool IsSuccessful { get; }
 
         public readonly string? ErrorMessage;
 
-        private readonly Solution? _newSolutionWithoutRenamedDocument;
-        private readonly (DocumentId documentId, string newName)? _renamedDocument;
+        public readonly (DocumentId documentId, string newName)? RenamedDocument;
 
         public readonly Solution? OldSolution;
 
         /// <summary>
         /// The final solution snapshot.  Including any renamed documents.
         /// </summary>
-        public readonly Solution? NewSolution;
+        public readonly Solution? NewSolutionWithoutRenamedDocument;
```

Comment:
I'm confused why we have both properties now -- it seems like the places you are using it would have been just the same as using NewSolutionWithRenamedDocument since there isn't a renamed document in the first place?